update.js

```diff
@@ -3,38 +3,42 @@
 const exec = require('child_process').execSync
 const fs = require('fs')
 
+const _ = require('lodash')
 const relative = require('require-relative')
 
 const extractDependency = require('./lib/extract-dependency')
 const updateShrinkwrap = require('./lib/update-shrinkwrap')
+const getValuesFromCI = require('./lib/get-values-from-ci')
 
 const pkg = relative('./package.json')
 
-const env = process.env
-
 module.exports = function update () {
   try {
     fs.readFileSync('./npm-shrinkwrap.json')
   } catch (e) {
     throw new Error('Without a shrinkwrap file present there is no need to run this script.')
   }
 
-  if (env.TRAVIS !== 'true') {
-    throw new Error('This script hast to run in an Travis CI environment')
+  const ciValues = getValuesFromCI()
+
+  if (_.isEmpty(ciValues)) {
+    throw new Error('This script must be run in a supported CI environment')
```

Comment:
`getValuesFromCI` could throw directly. Then we wouldn't need to copy this error message over to upload, too.